src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -105,3 +110,128 @@ void ShaderNetworkAlgo::removeUnusedShaders( ShaderNetwork *network )
 		}
 	}
 }
+
+namespace
+{
+
+const InternedString g_swizzleHandle( "swizzle" );
+const InternedString g_packHandle( "pack" );
+const InternedString g_inParameterName( "in" );
+const InternedString g_outParameterName( "out" );
+const InternedString g_packInParameterNames[3] = { "in1", "in2", "in3" };
+const boost::regex g_componentRegex( "^(.*)\\.([rgbxyz])$" );
```

Comment:
Because we're stuck needing to support GCC 4.8 for as long as we're either on legacy versions of certain DCCs or the current versions of the same DCCs don't adhere to VFXPlatform. And GCC 4.8's regex support is [somewhat lacking](https://stackoverflow.com/questions/12530406/is-gcc-4-8-or-earlier-buggy-about-regular-expressions). Here's one particularly illuminating quote from that link :

> Seriously though, who though that shipping an implementation of regex_search that only does "return false" was a good idea?